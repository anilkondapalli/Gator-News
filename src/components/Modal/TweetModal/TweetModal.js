import React, { useState } from "react";
import { Modal, Form, Button } from "react-bootstrap";
import "./TweetModal.scss"
import classNames from "classnames";
import { Camera } from "../../../utils/Icons";
import { toast } from "react-toastify";

import {addTweetApi} from "../../../api/tweet";

export default function TweetModal(props) {
    const { show, setShow } = props;
    const [message, setMessage] = useState("");
    const maxLength=280;


    const onSubmit = (e) => {
      e.preventDefault();
  
      if (message.length > 0 && message.length <= maxLength) {
        console.log("OK");
        addTweetApi(message)
          .then((response) => {
            console.log(response);
            if (response?.code >= 200 && response?.code < 300) {
              toast.success(response.message);
              setShow(false);
              //window.location.reload();
            }
          })
          .catch(() => {
            toast.warning("Error sending gator tweet, please try again later");
          });
      }
    };
  return (
    <Modal
      className="tweet-modal"
      show={show}
      onHide={() => setShow(false)}
      centered
      size="lg"
    >
        <Modal.Header>
        <Modal.Title>
          <Camera onClick={() => setShow(false)} />
        </Modal.Title>
      </Modal.Header>
      <Modal.Body>
      <Form onSubmit={onSubmit}>
          <Form.Control
            as="textarea"
            rows="6"
            placeholder="Whats Happening Gator?"
            onChange={(e) =>{
                e.preventDefault(); 
                setMessage(e.target.value);
                console.log(message);
            }
        }
          />
          <span
            className={classNames("count", {
              error: message.length > maxLength,
            })}
          >
            {message.length}
          </span>
          <Button role="submit"
            type="submit"
            disabled={message.length > maxLength || message.length < 1}
          >
              Buzz
              </Button>
          </Form>
          </Modal.Body>
    </Modal>
  );
}
