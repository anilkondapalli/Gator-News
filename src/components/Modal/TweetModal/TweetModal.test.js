import { render, screen } from '@testing-library/react';
import React from "react";
import TweetModal from './TweetModal';
import { create } from "react-test-renderer";

function Button(props) {
    return <button>Nothing to do for now</button>;
}

describe('TweetModal', () => {

    test("Render the TweetModal", () => {
        render(<TweetModal/>);
    });

    test("Match Snapshot", () => {
        const button = create(<Button />);
        expect(button.toJSON()).toMatchSnapshot();
      });


      test("Display expected text when clicked (testing the wrong way)", () => {
        const buttonComponent = create(<Button text="SUBSCRIBE TO BASIC" />);
        const buttonInstance = buttonComponent.getInstance();
        expect(buttonInstance).toBe(null);
      });

      test("It shows the expected text when clicked (testing the wrong way!)", () => {
        const component = create(<Button text="Buzz" />);
        const instance = component.root.props;
        expect(instance).toStrictEqual({"text": "Buzz"});
        console.log(instance);

       
      });
      test('Check Button Submission', () => {
        return screen.findByRole('submit')
          .then(() => {
            console.log('found');
          })
          .catch(() => {
            console.log('not found');
        });
    });

  });