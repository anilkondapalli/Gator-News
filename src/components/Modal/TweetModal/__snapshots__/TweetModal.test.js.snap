// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TweetModal Match Snapshot 1`] = `
<button>
  Nothing to do for now
</button>
`;

exports[`TweetModal Matches the snapshot 1`] = `
<button>
  Nothing to do for now
</button>
`;
